sciencer/providers/provider.py

```diff
@@ -39,3 +39,14 @@ def get_papers_by_author(self, author_id: str = "") -> List[Paper]:
         Returns:
             List[Paper]: The list of papers associated with an author.
         """
+
+    @abstractmethod
+    def get_paper_by_terms(self, terms: List[Paper], max_papers: int) -> List[Paper]:
```

Comment:
I think there is value added in making `max_papers` an optional argument.
This way, you keep flexibility and allow customization for whomever needs it